<launch>

  <param name="use_sim_time" value="true" />

  <arg name="rviz_display" default="true"/>

  <node name="lidar_undistortion_node" pkg="lidar_undistortion"  type="lidar_undistortion" output="screen">

    <param name="lidar_topic" value="/scan"/>
    <!-- Estimated time of lidar message transmission(ms) -->
    <param name="lidar_msg_delay_time" value="10.0"/>

    <param name="scan_direction_clockwise" value="true"/>

    <param name="imu_topic" value="/mti/sensor/imu"/>
    <param name="imu_frequency" value="100.0"/>

    <param name="output_frame_id" value="/laser"/>
    <param name="pub_raw_scan_pointcloud" value="true"/>

    <param name="pub_laserscan" value="true"/>
    <param name="laserscan_angle_increment" value="0.004"/>

    <param name="use_range_filter" value="true"/>
    <param name="range_filter_min" value="0.2"/>
    <param name="range_filter_max" value="12.0"/>

    <param name="use_angle_filter" value="true"/>
    <param name="angle_filter_min" value="-2.3"/>
    <param name="angle_filter_max" value="2.3"/>

    <param name="use_radius_outlier_filter" value="true"/>
    <param name="radius_outlier_filter_search_radius" value="0.1"/>
    <param name="radius_outlier_filter_min_neighbors" value="2"/>

  </node>

  <group if="$(arg rviz_display)">
     <node pkg="rviz" type="rviz" name="rviz"
       args="-d $(find lidar_undistortion)/rviz/lidar_undistortion.rviz" />
  </group>

</launch>
